### ISSUE TITLE
Add support for const values

### ISSUE THREAD

**Issue by hackaugusto** [0xPolygonMiden/miden-vm#634]
# Feature request

In some contexts it makes sense to use constants, e.g. value `-1` can be defined as:

```
push.0 sub.1
```

Which evaluates to `18446744069414584320`. Having a directive to name this value, e.g. `negative_one`, may make it easier to read the assembly.

2.a. It would also be interesting to define the value with const evaluation, like the value `-1` itself, just in case the field definition would ever change.

This issue was extracted from https://github.com/0xPolygonMiden/miden-vm/issues/633 . Relevant comment from @bobbinth : 

> The second makes sense too, but it is a bit tricky as -1 could mean different things as when interpreted as a field element vs. a u32 value. I think we could do something like specifying the type of a value like felt or u32.
> 
> More generally, I think naming values (i.e.,. declaring constants) could be super useful and probably deserves its own issue. For example, something like:
>
>> const.FOO.10
>>
>> being
>>    push.FOO # pushes 10 onto the stack
>> end
>
> Though, I don't love the const.NAME.value syntax. Maybe we can come up with something better without complicating the parser too much.

**Comment by hackaugusto** [comment]
Having this with a simple dead code elimination would also allow for some debugging with:

```
const.DEBUG.0

push.DEBUG if.true
  # debugging asserts
end
```

**Comment by bobbinth** [comment]
Closed by #692
